dynawo/sources/Models/Modelica/Dynawo/Electrical/Machines/OmegaRef/GeneratorSynchronous.mo

```diff
@@ -32,18 +32,18 @@ model GeneratorSynchronous "Synchronous machine"
   */
 
   // Voltage, active and reactive power at terminal
-  Types.VoltageModulePu UPu(start = U0Pu) "Voltage amplitude at terminal in pu (base UNom)";
-  Types.ActivePowerPu PGenPu(start = PGen0Pu) "Active power generated by the synchronous machine in pu (base SnRef)";
-  Types.ReactivePowerPu QGenPu(start = QGen0Pu) "Reactive power generated by the synchronous machine in pu (base SnRef)";
+  Types.VoltageModulePuConnector UPu(start = U0Pu) "Voltage amplitude at terminal in pu (base UNom)";
+  Types.ActivePowerPuConnector PGenPu(start = PGen0Pu) "Active power generated by the synchronous machine in pu (base SnRef)";
+  Types.ReactivePowerPuConnector QGenPu(start = QGen0Pu) "Reactive power generated by the synchronous machine in pu (base SnRef)";
   Types.ActivePower PGen(start = PGen0Pu*SystemBase.SnRef) "Active power generated by the synchronous machine in MW";
   Types.ReactivePower QGen(start = QGen0Pu*SystemBase.SnRef) "Reactive power generated by the synchronous machine in Mvar";
 
   // Output variables for external controllers
-  Dynawo.Connectors.ImPin UStatorPu(value(start = UStator0Pu)) "Stator voltage amplitude in pu (base UNom)";
+  Dynawo.Types.VoltageModulePuConnector UStatorPu(start = UStator0Pu) "Stator voltage amplitude in pu (base UNom)";
   Dynawo.Connectors.ImPin IStatorPu(value(start = IStator0Pu)) "Stator current amplitude in pu (base UNom, SnRef)";
   Dynawo.Connectors.ImPin QStatorPu(value(start = QStator0Pu)) "Stator reactive power generated in pu (base SnRef)";
   Dynawo.Connectors.ImPin QStatorPuQNom(value(start = QStator0PuQNom)) "Stator reactive power generated in pu (base QNomAlt)";
-  Dynawo.Connectors.ImPin IRotorPu(value(start = IRotor0Pu)) "Rotor current in pu (base SNom, user-selected base voltage)";
+  Dynawo.Types.CurrentModulePuConnector IRotorPu(start = IRotor0Pu) "Rotor current in pu (base SNom, user-selected base voltage)";
   Dynawo.Connectors.ImPin thetaInternal(value(start = ThetaInternal0)) "Internal angle in rad";
```

Comment:
Some ImPin connectors are preserved, shouldn't they be replaced by Types...Connector ?

Reply:
I only modified those which are required to compile existing preassembled

Reply:
Understood